src/games/strategy/engine/framework/GameRunner.java

```diff
@@ -222,212 +226,168 @@ private static void handleCommandLineArgs(final String[] args) {
         System.out.println("Unrecogized:" + arg1);
         if (!usagePrinted) {
           usagePrinted = true;
-          usage();
+          usage(gameMode);
         }
       }
     }
-    final String version = System.getProperty(TRIPLEA_ENGINE_VERSION_BIN);
-    if (version != null && version.length() > 0) {
-      final Version testVersion;
-      try {
-        testVersion = new Version(version);
-        // if successful we don't do anything
-        System.out.println(TRIPLEA_ENGINE_VERSION_BIN + ":" + version);
-        if (!ClientContext.engineVersion().getVersion().equals(testVersion, false)) {
-          System.out.println("Current Engine version in use: " + ClientContext.engineVersion());
-        }
-      } catch (final Exception e) {
-        System.getProperties().setProperty(TRIPLEA_ENGINE_VERSION_BIN, ClientContext.engineVersion().toString());
-        System.out.println(TRIPLEA_ENGINE_VERSION_BIN + ":" + ClientContext.engineVersion());
+
+    if (gameMode == GameMode.HEADLESS_BOT) {
```

Comment:
The diff gets pretty hard to follow here. The changes in this section are more easily reviewed commit-by-commit.